File: ABC120_B.py
T = list(map(int,input().split()))
A = T[0]
B = T[1]
K = T[2]
j = 0
num = []
for i in range(1,10000):
	if (A%i == 0) and (B%i == 0):
		num.insert(j,i)
		j= j+1
print (num[j-K])

File: ABC120_B.c
#include <stdio.h>

int main(){
	int A,B,K,j=1;
	int num[10000];
	scanf("%d %d %d",&A,&B,&K);
	for(int i=1;i<=10000;i++){
		if((A%i==0) && (B%i==0)){
			num[j] = i;
			j++;	
	}
}

printf("%d\n",num[j-K]);
	return 0;
}